### ISSUE TITLE
GraphQL API error after upstream schema updates

### ISSUE THREAD

**Issue by sio** [sio/cirrus-run#9]
Today all cirrun-run calls have started failing :

```
cirrus_run.api.CirrusAPIError: API returned 1 error(s):
[ { 'extensions': {'classification': 'ValidationError'},
    'locations': [{'column': 13, 'line': 2}],
    'message': 'Validation error of type FieldUndefined: Field '
               "'githubRepositories' in type 'Root' is undefined @ "
               "'githubRepositories'"}]
```

Most likely this is caused by backwards-incompatible changes to upstream GraphQL schema.

I haven't had a chance to look further yet.

**Comment by sio** [comment]
Looks like CirrusCI moves to become platform agnostic. All 'github*' schema members have been renamed to 'owner*' counterparts. Call signature has also been changed to incorporate new 'platform' parameter:

[Old:](https://github.com/cirruslabs/cirrus-ci-web/blob/1806cccc4498fc0fec46e3162e1caff6a41d21cb/schema.graphql#L423-L433)

```graphql
type Root {
  viewer: User
  repository(id: ID!): Repository
  githubRepository(owner: String!, name: String!): Repository
  githubRepositories(owner: String!): [Repository]
  githubOrganizationInfo(organization: String!): GitHubOrganizationInfo
  build(id: ID!): Build
  searchBuilds(repositoryOwner: String!, repositoryName: String!, SHA: String): [Build]
  task(id: ID!): Task
  webhookDelivery(id: String!): WebHookDelivery
}
```

[New:](https://github.com/cirruslabs/cirrus-ci-web/blob/97d6ac7dbddb42aaa9736c4caab79cf361540f2a/schema.gql#L922-L935)

```graphql
type Root {
  viewer: User
  repository(id: ID!): Repository
  ownerRepository(platform: String!, owner: String!, name: String!): Repository
  build(id: ID!): Build
  searchBuilds(repositoryOwner: String!, repositoryName: String!, SHA: String): [Build]
  task(id: ID!): Task
  hook(id: ID!): Hook
  webhookDelivery(id: String!): WebHookDelivery
  persistentWorkerPool(poolId: ID): PersistentWorkerPool
  persistentWorker(poolId: ID, name: String): PersistentWorker
  ownerInfo(platform: String, uid: ID): OwnerInfo
  ownerInfoByName(platform: String, name: String): OwnerInfo
}
```

**Comment by andreabolognani** [comment]
@sio I noticed this on Friday and didn't report it immediately in case it was a transient issue on Cirrus CI's part. I was going to report it now but I see you took care of everything already ;)

Can we expect a 1.0.1 release that includes this fix to land on PyPI soon?

Thank you for your continued work on cirrus-run :)

**Comment by sio** [comment]
Thanks for the kind words :) Yes, I plan to see this through and publish v1.0.1 to PyPI then.

---

Looks like I jumped the gun a bit with closing this issue.

Even though the query affected by schema update works correctly now, cirrus-run still can not create new builds via GraphQL API. Another error has surfaced.

I've trimmed the issue down to a minimal reproducible example, but it seems that I will not be able to proceed without some help from CirrusCI staff. See upstream issue here: https://github.com/cirruslabs/cirrus-ci-web/issues/426
